src/bentoml/_internal/frameworks/transformers.py

```diff
@@ -605,6 +643,337 @@ def make_default_signatures(pretrained: t.Any) -> ModelSignaturesType:
+                    proxies=proxies,
+                    revision=revision,
+                    variant=variant,
+                    **kwargs,
+                )
+                resolved_archive_file = cached_file(
+                    model_name_or_path,
+                    CONFIG_JSON_FILE_NAME,
+                    proxies=proxies,
+                    revision=revision,
+                    variant=variant,
+                )
+                src_dir = os.path.dirname(resolved_archive_file)
+            downloaded_model_successfully = True
+        except Exception:
+            logger.info(
+                "Failed to download model file for (%s). Will attempt to clone repository instead.",
+                model_name_or_path,
+            )
+
+        if not downloaded_model_successfully:
+            # clone entire repository as fallback
+            _try_import_huggingface_hub()
+            from huggingface_hub import snapshot_download
+
+            src_dir = snapshot_download(
+                model_name_or_path,
+                proxies=proxies,
+                revision=revision,
+            )
+
+            if sync_with_hub_version:
+                from huggingface_hub.file_download import REGEX_COMMIT_HASH
+
+                version = _extract_commit_hash(src_dir, REGEX_COMMIT_HASH)
+                if version is not None:
+                    tag.version = version
+
+    if task_name is not None:
+        # define a pipeline
+        if task_definition is not None:
+            from transformers.pipelines import get_supported_tasks
+
+            # NOTE: safe casting to annotate task_definition types
+            task_definition = t.cast(TaskDefinition, task_definition)
+
+            assert "impl" in task_definition, "'task_definition' requires 'impl' key."
+            options_args = (task_name, task_definition)
+
+            if task_name not in get_supported_tasks():
+                register_pipeline(task_name, **task_definition)
+                logger.info(
+                    "Task '%s' is a custom task and has been registered to the pipeline registry.",
+                    task_name,
+                )
+            assert (
+                task_name in get_supported_tasks()
+            ), f"Task '{task_name}' failed to register into pipeline registry."
+        else:
+            from transformers.pipelines import check_task
+            from transformers.pipelines import get_supported_tasks
+
+            try:
+                options_args = t.cast(
+                    "tuple[str, TaskDefinition]",
+                    check_task(task_name)[:2],
+                )
+                if task_name.startswith("translation"):
+                    options_args = (task_name, options_args[1])
+            except Exception:
+                raise BentoMLException(
+                    f"Task '{task_name}' is not a valid task for pipeline (available: {get_supported_tasks()}."
+                )
```

Comment:
Ye I'm still not sure why we need to check for pipeline in transformers here?

can you give me an example where a hf repo save with pipeline support?

Is it a new feature?

Reply:
I don't think hf repo saves with pipeline support but from what I'm aware of, our `save_model` method supports pipelines. So I assumed we wanted a similar behavior for `import_model` as well. Below is a comparison:

```python
# saving a pipeline
pipeline = transformers.pipeline("summarization", "t5-base")
bentoml.transformers.save_model("t5-base-summarization-pipeline", pipeline)

# importing to save as a pipeline
bentoml.transformers.import_model('t5-base-summarization-pipeline', 't5-base', task_name='summarization')

# service.py
bentoml.transformers.get("t5-base-summarization-pipeline").to_runner()
```

If this feature is not necessary, I can remove this 😃

Reply:
I think we have discussed this. `import_model` shouldn't load the model at all. using pipeline requires users to have the pipeline explicitly.

If users want to save pipeline, they can still use `save_model`.